Python/wrfrun/unCMIP6.py

```diff
@@ -0,0 +1,586 @@
+            outvarname = itm['aim_v']
+            lvltype = itm['type']
+            lvlmark = itm['lvlmark']
+            frq = itm['freq']            
+            if lvlmark == 'None':
+                lvlmark = ''
+            # File name
+            fn = input_root+'/'+varname+'_'+frq+lvlmark+'_'+self.model_name
+            fn += '_'+self.exp_id+'_'+self.esm_flag+'_'+self.grid_flag
+            if self.filestrdates[c]!='':
+                fn += '_'+self.filestrdates[c]+'.nc'
+            else:    
+                fn += '.nc'  
+            # Open dataset
+            ds = xr.open_dataset(fn)
+            # lat and lon handeling
+            if ((self.lat==None).all() and (self.lon==None).all()):
+                if ((varname=='tos') or (varname=='siconc') or (varname=='sithick')):
+                    raise ValueError("The first field in the table should be a variable different than tos, siconc or sithick.")
+                self.lat = np.array(ds.lat.values)
+                self.lon = np.array(ds.lon.values)
+            else:
+                if ((varname!='tos') and (varname!='sithick') and (varname!='siconc')):
+                    if not((np.allclose(np.array(ds.lat.values),self.lat,rtol=0.0,atol=1.0e-12)) and \
+                        (np.allclose(np.array(ds.lon.values),self.lon,rtol=0.0,atol=1.0e-12))):
+                        raise ValueError("Error: Inconsistent lat/lon values between different fields.")
+            # plev handeling for 3D fields
+            if lvltype=='3d':
+                if ((self.plev==None).all()):
+                    if (varname=='ua') or (varname=='va'):
+                        raise ValueError("The first 3D field in the table should be a variable different than ua and va.")
+                    self.plev = np.array(ds.plev.values)
+                else:
+                    if not(np.allclose(np.array(ds.plev.values),self.plev,rtol=0.0,atol=1.0e-12)):
+                        raise ValueError("Error: Inconsistent pressure levels between 3D fields.")            
+            # Find the soil layer, if needed
+            if (varname=='mrsol' or varname=='tsl'):
+                n_found = 0
+                for i in range(len(self.outsoillayers)):                    
+                    if outvarname[2:]==self.outsoillayers[i]:
+                        slvl = i
+                        n_found += 1
+                if not(n_found==1):
+                    raise ValueError("Error: Difficulty finding the soil layer.")               
+                # NOTE: This code assumes the same layers for soil moisture and temperature.            
+            # Read appropriate section of data
+            if (varname=='mrsol' or varname=='tsl'):            
+                ds[varname] = ds[varname].sel(depth=self.soildepths[slvl])
+            seltol = '20D' if itm['approx_dates'] else None
+            if (self.filestrdates[c]!=''):
+                if (varname=='mrsol' or varname=='tsl'):
```

Comment:
all these case distinctions seem very cumbersome and bloat the code... is this even necessary? do you ever refer back to it with the varname+str(slvl) name? and even if you do, you could just define a variable ds_varname that is either just varname or has the slvl extension, and then you make the case distinction ones and be done with it

Reply:
These last two comments need a bit of work. So for now I am leaving them.